### PR TITLE
fix: prevent file descriptors leaks to child processes

### PR DESCRIPTION
**Pull request by smira**
See #9412

I'll keep the issue open to track upstream PR status and remove replace directives.

### CODE REVIEW

**Comment by smira** [comment]
I used #9414 to verify fixes

File: go.mod [hunk 1]
```diff
@@ -26,6 +26,18 @@ replace (
 	gopkg.in/yaml.v3 => github.com/unix4ever/yaml v0.0.0-20220527175918-f17b0f05cf2c
 )
 
+// fd-leak related replacements: https://github.com/siderolabs/talos/issues/9412
```
Comment (dsseng) [inline go.mod:29]: Maybe leave a TODO/FIXME to not forget to replace these versions with upstream ones once the fixes are upstreamed. Maybe we can just leave the issue open instead

**Review by dsseng — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (smira) [reply go.mod:29]: that's what I'm doing: 

> I'll keep the issue open to track upstream PR status and remove replace directives.

**Review by Unix4ever — APPROVED** [review]

**Review by dsseng — APPROVED** [review]
reviewed after force-push

**Pull request merged**